### ISSUE TITLE
[Bug]: Longchat-7b-v1.5-32k get different result with TP4 and TP2

### ISSUE THREAD

**Issue by sitabulaixizawaluduo** [vllm-project/vllm#4076]
### Your current environment

```text
The output of `python collect_env.py`
```
GPUS: L40
vllm version: 0.4.0 post

### 🐛 Describe the bug

```
model = LLM(model='longchat-7b-v1.5-32k',
                       trust_remote_code=True,
                       tensor_parallel_size=2)
## len(prompts) == 32000
sampling_params = SamplingParams(temperature=0, max_tokens=512)
ouputs = model.generate(prompts, sampling_params=sampling_params)
print(outputs[0].outputs[0].text)
```
when  I set tensor_parallel_size to 2, result is **"This report provides an overview of multiyear procurement (MYP) and block buy contracting (BBC) mechanisms, which are special contracting approaches permitted by the Department of Defense (DOD) for a limited number of defense acquisition programs. These mechanisms aim to reduce weapon procurement costs by a few or several percent compared to the standard annual contracting approach. The report discusses the potential issues for Congress concerning the future use of MYP and BBC, including whether to use them more frequently, less frequently, or about as frequently as they are currently used. It also addresses whether a permanent statute should be established to govern the use of BBC, similar to the permanent statute for MYP.

The report highlights the potential benefits of using MYP and BBC, such as promoting efficiency and reducing procurement costs. However, it also emphasizes the risks of using these mechanisms, including reduced flexibility for making changes in procurement programs in future years and potential confusion with other contracting approaches.

The report further discusses the differences between MYP, BBC, and contracts with options, as well as the potential use of these mechanisms in the Coast Guard. It concludes by summarizing the key provisions of 10 U.S.C. 2306b, which governs the use of MYP and BBC."**
when I set  tensor_parallel_size to 4, result is **"This report provides an overview of multiyear procurement (MYP) and block buy contracting (BBC), two special contracting mechanisms permitted by the Department of Defense (DOD) for a limited number of defense acquisition programs. These mechanisms aim to reduce weapon procurement costs by a few or several percent compared to the standard annual contracting approach. The report discusses the potential issues for Congress concerning the future use of MYP and BBC, the possibility of creating a permanent statute to govern the use of BBC, and whether the Coast Guard should begin making use of these mechanisms.

The report also highlights the potential savings from using MYP and BBC, the differences between these contracting mechanisms and annual contracting, and the potential disadvantages of using these mechanisms on flexibility for making procurement changes. It explains the criteria for programs to qualify for MYP and the process for approval by Congress.

The report further discusses the use of MYP and BBC in the past, the potential benefits and drawbacks of using these mechanisms more frequently, and the potential creation of a permanent statute to govern the use of BBC. It also examines the possibility of the Coast Guard using these mechanisms and the potential savings from using them in the Coast Guard's ship and aircraft acquisition programs.

In conclusion, the report provides valuable information on the use of MYP and BBC in defense acquisition programs and their potential impact on procurement costs and flexibility. It also raises important questions for Congress to consider when deciding whether to use these mechanisms more frequently or create a permanent statute to govern their use."**
